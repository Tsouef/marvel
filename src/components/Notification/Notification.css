.notification {
  position: fixed;
  top: 0;
  left: 50%;
  padding: 0 20px;
  border-radius: 3px;
  transform: translate(-50%, 0);
  z-index: 2;
}

.notification.success {
  background-color: rgba(51, 175, 100, 1);
}

.notification.error {
  background-color: red;
}
