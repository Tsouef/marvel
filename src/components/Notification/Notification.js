import React from 'react';
import { connect } from 'react-redux';
import './Notification.css';

const Notification = ({ favorites }) => {
  if (favorites.showNotification) {
    return (
      <div className="notification">
        <p>{favorites.notificationContent}</p>
      </div>
    );
  }

  return null;
};

function mapStateToProps({ favorites }) {
  return { favorites };
}

export default connect(mapStateToProps, null)(Notification);
