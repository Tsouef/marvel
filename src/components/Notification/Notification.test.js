import React from 'react';
import renderer from 'react-test-renderer';

import { Notification } from './Notification';

describe('Notification Component', () => {
  it('should render message success correctly', () => {
    const props = {
      favorites: {
        showNotification: true,
        notificationState: 'success',
        notificationContent: 'Message success'
      }
    };
    const tree = renderer.create(<Notification {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render message error correctly', () => {
    const props = {
      favorites: {
        showNotification: true,
        notificationState: 'error',
        notificationContent: 'Already use'
      }
    };
    const tree = renderer.create(<Notification {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should no render message', () => {
    const props = {
      favorites: {
        showNotification: false,
        notificationState: 'error',
        notificationContent: 'Already use'
      }
    };
    const tree = renderer.create(<Notification {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
