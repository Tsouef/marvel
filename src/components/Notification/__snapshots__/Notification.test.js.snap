// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification Component should no render message 1`] = `null`;

exports[`Notification Component should render message error correctly 1`] = `
<div
  className="notification error"
>
  <p>
    Already use
  </p>
</div>
`;

exports[`Notification Component should render message success correctly 1`] = `
<div
  className="notification success"
>
  <p>
    Message success
  </p>
</div>
`;
